### PR TITLE
Move MetaSkeleton-dependent problems to aikido::planner::dart.

### PR DESCRIPTION
**Pull request by brianhou**
Some problem types are associated with a `MetaSkeleton` (`ConfigurationToTSR`, `ConfigurationToEndEffector[Offset|Pose]`). However, the `MetaSkeleton` that we use to solve those problems can't be stored in the problem itself because we currently take `Problem`s as `const`.

@jslee02 and I decided that it would make sense for `Planner`s that solve these problems to take in a non-const `MetaSkeleton` in the constructor. This PR introduces the `aikido::planner::dart` namespace for these problem types.

In a future PR, we should add corresponding `SingleProblemPlanner`s for these problems. It would also be useful to have an `aikido::planner::dart::ConfigurationToConfigurationPlanner` that takes in a non-const `MetaSkeleton` in its constructor so that we can adapt them more easily.

***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [x] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: src/planner/vectorfield/MoveEndEffectorOffsetVectorField.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ MoveEndEffectorOffsetVectorField::MoveEndEffectorOffsetVectorField(
 bool MoveEndEffectorOffsetVectorField::evaluateCartesianVelocity(
     const Eigen::Isometry3d& pose, Eigen::Vector6d& cartesianVelocity) const
 {
-  using dart::math::logMap;
+  using ::dart::math::logMap;
```
Comment (sniyaz) [inline src/planner/vectorfield/MoveEndEffectorOffsetVectorField.cpp]: Not really a review thing but more a question from somebody who doesn't know C++ as well- what is the purpose of adding `::` here?

**Comment by sniyaz** [comment]
This personally looks good to me 👍

File: src/planner/vectorfield/MoveEndEffectorOffsetVectorField.cpp (hunk 1 above)
Comment (jslee02) [reply src/planner/vectorfield/MoveEndEffectorOffsetVectorField.cpp]: The leading `::` is used to represent the following namespace is a top-level namespace. We use this when there is the same name namespace in the "active" namespace. In our case, we use it to distinguish `::dart` (i.e., the DART's top-level namespace) and `::aikido::statespace::dart`/`::aikido::planner::dart`.

File: src/planner/dart/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+set(sources
```
Comment (jslee02) [inline src/planner/dart/CMakeLists.txt]: I believe we only make a new sub-component when it requires optional 3rd-party dependencies. For example, other dart subdirectories of `statespace` and `constraint` are not components but included its parent components.

In this sense, I think `aikido_planner_ompl` and `aikido_planner_vectorfield` also shouldn't be sub-components.

**Pull request merged**